pkg/api/handlers/compat/containers_stats.go

```diff
@@ -174,11 +189,13 @@ func StatsContainer(w http.ResponseWriter, r *http.Request) {
 			ID:       stats.ContainerID,
 			Networks: net,
 		}
-
-		utils.WriteJSON(w, http.StatusOK, s)
-		if flusher, ok := w.(http.Flusher); ok {
-			flusher.Flush()
+		// encode data
+		if err := enc.Encode(s); err != nil {
+			utils.Error(w, "failed stat encoding", http.StatusBadRequest, err)
```

Comment:
You have already send the StatusOK so you cannot send StatusBadRequest here.

Reply:
what should i use @jwhonce

Reply:
Most you can do is log error.